src/tezWrapper.ml

```diff
@@ -0,0 +1,279 @@
+  let state, ops = tez_wrapper_fa2_run_transfer (state, xs) in (* TODO: Uses specialized tez_wrapper_fa2_run_transfer *)
+  (ops, state)
+
+let[@inline] update_operators (state: tez_wrapper_state) (xs: fa2_update_operator list) : LigoOp.operation list * tez_wrapper_state =
+  let _ = ensure_no_tez_given () in
+  let state = { state with fa2_state = fa2_run_update_operators (state.fa2_state, xs) } in (* NOTE: No need for specialized calls, since the spec does not require checking the token_ids here. *)
+  (([]: LigoOp.operation list), state) (* no need to originate vaults *)
+
+(*****************************************************************************)
+(**                      {1 WRAPPER ENTRYPOINTS}                             *)
+(*****************************************************************************)
+
+let[@inline] deposit (state: tez_wrapper_state) (_: unit) : LigoOp.operation list * tez_wrapper_state =
+  (* 1. Update the balance on the ledger *)
+  let state_fa2_state = ledger_issue_tez_token (state.fa2_state, !Ligo.Tezos.sender, !Ligo.Tezos.amount) in
+  let state = { state with fa2_state = state_fa2_state } in
+  (* 2. Create a vault, if it does not exist already, and update the map. *)
+  let op_opt, state, vault_address = find_vault_address state !Ligo.Tezos.sender in
+  (* 3. Transfer the actual tez to the vault of the sender. *)
+  let op = match (LigoOp.Tezos.get_entrypoint_opt "%vault_receive_tez" vault_address : unit Ligo.contract option) with
+    | Some c -> LigoOp.Tezos.unit_transaction () !Ligo.Tezos.amount c
+    | None -> (Ligo.failwith error_GetEntrypointOptFailureVaultReceiveTez : LigoOp.operation) in
+  match op_opt with
+  | None -> ([op], state)
+  | Some origination -> ([origination; op], state)
+
+let[@inline] withdraw (state: tez_wrapper_state) (amnt: Ligo.tez) : LigoOp.operation list * tez_wrapper_state =
+  (* 1. Ensure no tez given *)
+  let _ = ensure_no_tez_given () in
+  (* 2. Reduce the balance of the tez owner *)
+  let state_fa2_state = ledger_withdraw_tez_token (state.fa2_state, !Ligo.Tezos.sender, amnt) in
+  let state = { state with fa2_state = state_fa2_state; } in
+  (* 3. Create a vault, if it does not exist already, and update the map. *)
+  let op_opt, state, vault_address = find_vault_address state !Ligo.Tezos.sender in
+  (* 4. Instruct the vault to send the actual tez to the owner *)
+  let op = match (LigoOp.Tezos.get_entrypoint_opt "%vault_send_tez_to_contract" vault_address : (Ligo.tez * Ligo.address) Ligo.contract option) with
+    | Some c -> LigoOp.Tezos.tez_address_transaction (amnt, !Ligo.Tezos.sender) (Ligo.tez_from_literal "0mutez") c
+    | None -> (Ligo.failwith error_GetEntrypointOptFailureVaultSendTezToContract : LigoOp.operation) in
+  match op_opt with
+  | None -> ([op], state)
+  | Some origination -> ([origination; op], state)
+
+let[@inline] set_delegate (state: tez_wrapper_state) (kho: Ligo.key_hash option) : LigoOp.operation list * tez_wrapper_state =
```

Comment:
I'm not sure if we need to create the vault if it doesn't exist when calling this entrypoint, but it does at least make the behavior consistent with the FA2 entrypoints

Reply:
That's a fair point, we could also fail without any problems. I mostly tried to make vault origination implicit (and eager) everywhere, so that, indeed, similarly to FA2, operations work fine even if an account is not created yet or if a user has zero tez, etc. I guess this approach is rather eager when it comes to vault creation, but I think that's actually nice: we never have to fail due to "vault non-existent" errors. I can't think of a downside at the moment but perhaps we shoudl revisit the approach.